### PR TITLE
Search config

### LINKED ISSUE #53: Pass Solr-endpoint query suffix as options to the Smarti widget
Rocket.Chat Version: 
Running Instances: 
DB Replicaset OpLog:
Node Version:

### PR DESCRIPTION
**Pull request by ruKurz**
@RocketChat/dbs-ai

Closes #53

### CODE REVIEW

File: packages/dbs-ai/client/views/app/tabbar/smarti.js [hunk 1]
```diff
@@ -38,13 +38,30 @@ Template.dbsAI_smarti.onRendered(function() {
...
 
-			self.smarti = new window.SmartiWidget(self.find('.external-message'), {
+			const smartiOptions = {
 				socketEndpoint: WEBSOCKET_URL,
 				smartiEndpoint: DBS_AI_Redlink_URL,
 				channel: self.data.rid,
-				inputCssSelector: '.autogrow-shadow'
-			});
+				inputCssSelector: '.autogrow-shadow',
+				widget:{
```
Comment (mrsimpson) [inline packages/dbs-ai/client/views/app/tabbar/smarti.js]: @ruKurz the css-selector is also an option for the widget - actually everything in the options is related to the widget. Thus, I'd not introduce a widget-property.

File: packages/dbs-ai/client/views/app/tabbar/smarti.js [hunk 2]
```diff
@@ -38,13 +38,30 @@ Template.dbsAI_smarti.onRendered(function() {
...
+					'query.dbsearch.keyword': {
```
Comment (mrsimpson) [inline packages/dbs-ai/client/views/app/tabbar/smarti.js]: @ruKurz is `query.dbsearch.keyword` really needed? what is the semantic of the sub-property `keyword`? I thought it was a separate query provider.

File: packages/dbs-ai/client/views/app/tabbar/smarti.js [hunk 3]
```diff
@@ -38,13 +38,30 @@ Template.dbsAI_smarti.onRendered(function() {
...
+						numOfRows:2,
+						suffix: customSuffix
+					},
+					'query.dbsearch': {
```
Comment (mrsimpson) [inline packages/dbs-ai/client/views/app/tabbar/smarti.js]: @ruKurz the multiple dots imply a composition - why not have that as real sub-object-properties?
wydt about the following:
```javascript
const smartiOptions = {
				socketEndpoint: WEBSOCKET_URL,
				smartiEndpoint: DBS_AI_Redlink_URL,
				channel: self.data.rid,
				inputCssSelector: '.autogrow-shadow',
				queries:[
					{
				             dbsearch: {
						  numOfRows:2,
						  suffix: customSuffix
				              },
                                             keyword: {
						  disabled: true
				              }
					}
				],
				lang:'de'
			};
```

**Review by mrsimpson — COMMENTED** [review]
@ruKurz I have made some suggestions about the configuration object to be passed to the widget.
Since this format also affects the smarti-side, it has an impact to the smarti-PR, thus: @tkurz what do you tink about the format?

**Comment by mrsimpson** [comment]
@ruKurz please lint the source before checking in. It was really nice to see Mr. Travis happy ;) https://travis-ci.org/mrsimpson/Rocket.Chat/builds/255809955#L2309

**Comment by ruKurz** [comment]
@mrsimpson all of the options are related to the Smarti widget constructor. So IMHO all your comments should be placed there. And if the Smarti widget/interface is "fine", we can change it here.

@mrsimpson the lint error is not part of this PR.

**Pull request merged**